scripts/tests/test_zkai_paper_claim_pack_gate.py

```
@@ -0,0 +1,75 @@
+            gate.write_json(path, self.payload)
+            loaded = json.loads(path.read_text(encoding="utf-8"))
+            self.assertEqual(loaded, self.payload)
+
+    def test_write_json_rejects_symlink_output(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            tmp_path = gate.pathlib.Path(tmp)
+            target = tmp_path / "target.json"
+            target.write_text("{}", encoding="utf-8")
+            link = tmp_path / "link.json"
+            link.symlink_to(target)
+            with self.assertRaisesRegex(gate.ClaimPackGateError, "output path must not be a symlink"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)